runtime/msgpack-serialization.h

```diff
@@ -70,6 +70,20 @@ inline Optional<string> f$instance_serialize(const class_instance<InstanceClass>
   return result;
 }
 
+template<class InstanceClass>
+inline string f$instance_serialize_safe(const class_instance<InstanceClass> &instance) noexcept {
+  vk::msgpack::packer_float32_decorator::clear();
+  vk::msgpack::CheckInstanceDepth::depth = 0;
+  string err_msg;
+  auto result = f$msgpack_serialize(instance, &err_msg);
+  if (vk::msgpack::CheckInstanceDepth::is_exceeded()) {
+    THROW_EXCEPTION(new_Exception(string(__FILE__), __LINE__, string("maximum depth of nested instances exceeded")));
```

Comment:
`THROW_EXCEPTION` macro doesn't return from the current function, you need to do it manually. You can check other usages